### PR TITLE
Add ES2017 string padding

### LINKED ISSUE #11942: Missing es2017 String Padding Declaration
**es2017** target missing declaration for **Stage 4** String Padding [proposal](https://github.com/tc39/proposal-string-pad-start-end):
- [padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)
- [padEnd](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd)

---

**TypeScript Version:**  2.0.6 && nightly (2.1.0-dev.20161029)

### PR DESCRIPTION
**Pull request by imcotton**
Fixes #11942

### CODE REVIEW

File: tests/cases/conformance/es2017/useStringPadding3.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// @target: es6
```
Comment (mhegazy) [inline tests/cases/conformance/es2017/useStringPadding3.ts]: the tests need baselines, so travis is going to fail. you need to run `jake baseline-accept` after running the tests to include the baselines.

For this change, i do not think you need the tests. so i would leave them out.

**Comment by mhegazy** [comment]
otherwise looks good. thanks!

**Comment by imcotton** [comment]
I mostly following changeset in #8538, however running `jake baseline-accept` I've got zero diffs under `tests/baselines/reference`, couldn't figure out why.

If I understand correctly, then I'd just remove these `useStringPadding[1-4].ts` and enough to make the cut, right?

**Comment by mhegazy** [comment]
> 've got zero diffs under tests/baselines/reference, couldn't figure out why.

did you run the tests first? you should run the tests by `jake runtests t=useStringPadding`, they should fail, since there are no baselines to compare to. the running `jake baseline-accept` would copy the baselines generated from the failed test run into the references folder.

> If I understand correctly, then I'd just remove these useStringPadding[1-4].ts and enough to make the cut, right?

I would say the tests are not required. We have no test coverage for most of the library functions, and we do not change them once they are added anyways. so feel free to remove them.

**Comment by imcotton** [comment]
Thank you for the guidance, I've managed generate baselines after all, but since it's trivial therefore I've removed them from commits, and travis happy now.

**Comment by jhpratt** [comment]
@mhegazy Still running into this issue on TS 2.8.3.

`Property 'padStart' does not exist on type 'string'.`

**Comment by jhpratt** [comment]
Ah, I had brought the target up to current (2018) just to test, but didn't think of the lib. Moving that from es6 to es2017 fixed it. Thanks!

**Pull request merged**